### PR TITLE
feat!: backport query changes from main

### PR DESCRIPTION
**Pull request by clason**
This PR backports (most) query changes from `main` that are compatible with Neovim 0.9.2 (the current minimum version). Biggest changes:

* Highlight captures have been renamed across the board to be more compatible with upstream and Helix (see commit message for details); this will require colorscheme changes. I will upstream these changes to Neovim, so breaking changes are unlikely in the future.
* Use upstream format (`local.scope` etc.) for locals as well.
* Backport the query formatter (🥳 )
* Add the `printf` parser that was erroneously added to `main` instead of `master`.

## Roadmap

I'll leave this up for a bit for review feedback and increased visibility. Over the course of this week, I'll

1. add a 0.9.2 tag for people who don't want to deal with the colorscheme changes (yet)
2. merge this PR
3. upstream the highlight capture names to Neovim https://github.com/neovim/neovim/pull/27067
4. (manually, painfully) rebase `main` on top of these changes
5. start merging the (ready) PRs that were blocked by these changes

## Commits 
- feat(locals)!: switch to upstream captures
- feat(injections): add printf format strings
- chore(tests): consistent captures
- feat!: align standard captures with upstream
- test(queries): print ALL errors at end
- feat: query formatting script
- refactor: manual pre-cleanup
- ci: add query lint job
- chore: query formatting
- docs: update CONTRIBUTING.md with formatting
- chore(rust): tweak injections, add json macro injection
- feat(git_config): add missing injections
- fix(highlights): improve consistency
- fix(highlights): improve member/property distinction
- fix(css): make the highlights more sensible

### CODE REVIEW

**Review by lucario387 — APPROVED** [review]

**Comment by lucario387** [comment]
`CONTRIBUTING.md` is missing a section about formatting.

**Comment by clason** [comment]
Yes, that part didn't apply cleanly.

File: CONTRIBUTING.md
Comment (ObserverOfTime) [inline CONTRIBUTING.md:90]: GitHub won't highlight the blocks with `query`, it needs to be `scheme`.

File: queries/cpon/highlights.scm
Comment (ObserverOfTime) [inline queries/cpon/highlights.scm:50]:
```suggestion
(("\"" @conceal)
```

File: CONTRIBUTING.md
Comment (clason) [reply CONTRIBUTING.md:90]: But Neovim will ;) And you know which side of _my_ bread is buttered ;)

File: queries/markdown/highlights.scm
Comment (ObserverOfTime) [inline queries/markdown/highlights.scm:97]: These should all be `@markup.list` (even though they're commented out).

File: CONTRIBUTING.md
Comment (ObserverOfTime) [reply CONTRIBUTING.md:90]: > But Neovim will ;) And you know which side of *my* bread is buttered ;)

Yeah, but we can have *both*.

> (Maybe this will get them to start :))

It won't. :/ (github-linguist/linguist#5746)

Comment (clason) [reply CONTRIBUTING.md:90]: > Yeah, but we can have both.

No we can't, because the proper highlighting (modelines and all) only works for the `query` language.

File: queries/cpon/highlights.scm
Comment (clason) [reply queries/cpon/highlights.scm:50]: Please only comment on **backporting** mistakes. 

If there are changes you'd like to make on `main`, just open a PR and I'll merge and cherry-pick the commits for this PR.

File: CONTRIBUTING.md
Comment (ObserverOfTime) [reply CONTRIBUTING.md:90]: Better to have 90% in both cases than 100% in one and 0% in the other.

File: .github/workflows/lint.yml
Comment (lucario387) [inline .github/workflows/lint.yml:42]: Is it intended to use nightly for this :thinking:

File: CONTRIBUTING.md
Comment (lucario387) [inline CONTRIBUTING.md:270]: Typo I think

File: CONTRIBUTING.md
Comment (clason) [reply CONTRIBUTING.md:90]: The only thing you get on Github is slightly grayer comments; I count that as "90%" ;)

File: .github/workflows/lint.yml
Comment (clason) [reply .github/workflows/lint.yml:42]: Yes, why not? (I've adjusted so the script works on 0.9.x, but that will give us some early warning if things break.)

File: queries/cpon/highlights.scm
Comment (clason) [reply queries/cpon/highlights.scm:50]: Or wait until this is merged and fix it on `master` -- the less I have to rebase, the better :)

File: queries/rust/injections.scm
Comment (lucario387) [inline queries/rust/injections.scm:1]: Just to be sure, have you re-apply the `injection.combined` for `main` branch ?

Comment (clason) [reply queries/rust/injections.scm:1]: Yes, the goal is that when I'm done rebasing, the HEAD of master should have the exact same queries as now. (Which will be tricky.)

**Comment by lucario387** [comment]
Just to note: actual minimal stable version that can use nvim-treesitter is v0.9.2 (empty injections fix)

**Comment by clason** [comment]
> Just to note: actual minimal stable version that can use nvim-treesitter is v0.9.2 (empty injections fix)

We should document this then. Does this mean that more changes from `main` can be backported?

**Comment by lucario387** [comment]
I don't think so. The backport for empty-range injections has been done already.

File: tests/query/injections/html/test-html-injections.html
Comment (lucario387) [inline tests/query/injections/html/test-html-injections.html:17]: I'd say it's lucky that despite all `@..` is moved 1 space to the right here, it's still inside the intended range.

File: CONTRIBUTING.md
Comment (lucario387) [inline CONTRIBUTING.md]: This is later changed in e33e9461d51d43b59c1d983af901aa15bc103d35, but I think it should be changed in this commit as well

File: tests/query/injections/html/test-html-injections.html
Comment (clason) [reply tests/query/injections/html/test-html-injections.html:17]: Not lucky, tested ;)

**Review by lucario387 — APPROVED** [review]
Just went through all the diffs. It's like this
```diff
---
> diff --git a/runtime/queries/yaml/highlights.scm b/runtime/queries/yaml/highlights.scm
1105,1106c1118,1119
< --- a/queries/yaml/highlights.scm
< +++ b/queries/yaml/highlights.scm
---
> --- a/runtime/queries/yaml/highlights.scm
> +++ b/runtime/queries/yaml/highlights.scm
1154c1167
< diff --git a/queries/yang/highlights.scm b/queries/yang/highlights.scm
---
> diff --git a/runtime/queries/yang/highlights.scm b/runtime/queries/yang/highlights.scm
1156,1157c1169,1170
< --- a/queries/yang/highlights.scm
< +++ b/queries/yang/highlights.scm
---
> --- a/runtime/queries/yang/highlights.scm
> +++ b/runtime/queries/yang/highlights.scm
```

So only filename diffs, nothing changes compared to main branch

File: CONTRIBUTING.md
Comment (clason) [reply CONTRIBUTING.md]: I don't think that's easily possible, unless I either squash commits that shouldn't be squashed or completely re-stage all these changes. (Remember that I need to rebase `main` on top of this PR, and that is easier if the commits are cherry-picked exactly.)

Comment (lucario387) [reply CONTRIBUTING.md]: Then i'm fine with the current revision. Also easier to view diffs with `main`

**Pull request merged**
